metricbeat/module/kubernetes/state_statefulset/_meta/data.json

```diff
@@ -1,30 +1,30 @@
 {
-  "@timestamp": "2017-05-10T16:46:37.821Z",
-  "beat": {
-    "hostname": "X1",
-    "name": "X1",
-    "version": "6.0.0-alpha1"
-  },
-  "kubernetes": {
-    "namespace": "jenkins",
-    "statefulset": {
-      "name": "wise-lynx-jenkins-1616735317",
-      "created": 123454,
-      "replicas": {
-        "desired": 1,
-        "observed": 1,
-      },
-      "generation": {
-        "desired": 1,
-        "observed": 1,
-      }
+    "@timestamp": "2019-03-01T08:05:34.853Z",
+    "event": {
+        "dataset": "kubernetes.statefulset",
+        "duration": 115000,
+        "module": "kubernetes"
+    },
+    "kubernetes": {
+        "namespace": "custom",
+        "statefulset": {
+            "created": 1511999697,
+            "generation": {
+                "desired": 5,
+                "observed": 3
+            },
+            "name": "mysql",
+            "replicas": {
+                "desired": 6,
+                "observed": 3
+            }
+        }
+    },
+    "metricset": {
+        "name": "state_statefulset"
+    },
+    "service": {
+        "address": "127.0.0.1:55555",
+        "type": "kubernetes"
     }
-  },
-  "metricset": {
-    "host": "192.168.99.100:18080",
-    "module": "kubernetes",
-    "name": "state_statefulset",
-    "namespace": "statefulset",
```

Comment:
@exekias We seem to loose this field here but I would argue it's not an issue. WDYT?

Reply:
Is this the kubernetes namespace? that field would be needed to identify the statefulset we are referring to

Reply:
nevermind, I see this is a different namespace, we can drop it :+1:

Reply:
You'll see this "dropping field" pattern in more metricsets in Kubernetes. Just to give you some context @exekias the main idea was to achieve the migration, even dropping few things that weren't critical. In this case I have to admit that I was getting confused with the `namespace` thing (it seems there are 2 namespaces, the one of kubernetes and _the other_ :smile: ) but I'd need to change quite a lot of code to test the module fields and the root fields (current code _only_ tests metricset fields)

So, now that we have the new testing framework of @ruflin it seemed like the way to go and maybe delete current testing code if it doesn't achieve much more than what we get with the contents of `testdata` folder.

I know it seems confusing so just ping me if you need a longer explanation :wink:

Reply:
I think I got it, I'm ok with the change :+1: